framework/selftest.cpp

```diff
@@ -179,6 +180,70 @@ static int selftest_logs_getcpu_run(struct test *test, int cpu)
     return EXIT_SUCCESS;
 }
 
+static int selftest_logs_reschedule_init(struct test *test)
+{
+    // In order to always get the same result and avoid race conditions,
+    // we use semaphores to synchronize the access to reschedule()
+    int sem_size = num_cpus() - 1;
+    sem_t *reschedule_sem = (sem_t *) malloc(sizeof(sem_t)*sem_size);
```

Comment:
`calloc(sem_size, sizeof(sem_t))`

Alternatively, you could use `std::binary_semaphore`. I've never used it, but it supposed to work.
```c++
    test->data = new std::binary_semaphore[sem_size];
// in cleanup:
    delete[] static_cast<std::binary_semaphore *>(test->data);
```